creator-node/test/lib/helpers.js

```diff
@@ -0,0 +1,73 @@
+const request = require('supertest')
+const fs = require('fs')
+const uuid = require('uuid/v4')
+const path = require('path')
+
+const DiskManager = require('../../src/diskManager')
+
+const { wait } = require('./utils')
+const { handleTrackContentRoute } = require('../../src/components/tracks/tracksComponentService')
+
+const uploadTrack = async (filePath, cnodeUserUUID, ipfs, blacklistManager) => {
+  const { fileUUID, fileDir } = saveFileToStorage(filePath)
+  const resp = await handleTrackContentRoute({
+    logContext: {
+      requestID: uuid(),
+      requestMethod: 'POST',
+      requestHostname: '127.0.0.1',
+      requestUrl: '/track_content'
+    }
+  },
+  {
+    fileName: `${fileUUID}.mp3`,
+    fileDir,
+    fileDestination: fileDir,
+    session: {
+      cnodeUserUUID
+    }
+  },
+  ipfs,
+  blacklistManager
+  )
+
+  return resp.object
+}
+
+const saveFileToStorage = (filePath) => {
```

Comment:
Just curious why do this vs hit the route directly?

Reply:
hitting the route directly was causing issues with the mock, e.g. the mock was not being respected properly. to go around that, I explicitly called mock the middleware to upload a file, and then call `handleTrackContentRoute` directly

Reply:
What was the exact issue? I think it would be worthwhile having the route level test if we can figure out the mock thing

Reply:
basically, we needed to [update the required module to bet the mock registry ](https://github.com/AudiusProject/audius-protocol/pull/1452/files#diff-f3bcf7dd62a4ef56c46dbc814c34477933eb6459f8bd8c99bfd42c35acab36f7R28-R31) for the app to use the mockServiceRegsitry. however, with this approach, we also need to set the mock requirements, like the number of times the mock is called, in this code path, like:

```
require.cache[require.resolve('../../src/serviceRegistry')].exports.serviceRegistry.ipfs.addFromFS.exactly(33)
```

this implied needing to update each test to specify how many times the mock was called / moving this logic out of beforeEach() and into each individual method. seemed too obtrusive of a change. also, I think by testing the app logic of this route, we can use mocks more easily.

Reply:
i've spent some time looking into ways we can hit the route, however, could not find a solution that would not require a big refactor

Reply:
testing the app logic directly was the simplest and most effective. you can see that I left some tests in this PR that still hit `/track_content` route. this was to test the middleware

Reply:
I agree with the decision choice @vicky-g made here. Testing the business logic "meat & potatoes" is much more important than testing that the endpoint will respond--it's also the same direction the healthCheckController took, etc.

Reply:
Fair enough, we can resolve this

Reply:
I do think there is value in hitting the actual route. I think we can do so via mad-dog later